### DISCUSSION TITLE
Is it possible to send a message to nemo, maybe through d-bus, to select a specific file, given its path?

### DISCUSSION THREAD

**Discussion by anaximeno** [linuxmint/discussions#379] [Development]
For example if I were in the folder below:

![Captura de ecrã de 2024-04-05 14-39-38](https://github.com/linuxmint/discussions/assets/63073056/b2d6f294-3988-4e43-926d-a895d296ac80)

I would like to send a message to nemo with the path of the folder named **templates**, and the result would be:

![Captura de ecrã de 2024-04-05 14-39-56](https://github.com/linuxmint/discussions/assets/63073056/8dc6602f-d0c9-4f78-af94-44523436a028)

This could be useful in some actions ideas/improvements I have.

**Comment by rcalixte** [comment]
This is already supported by Nemo. You just need to pass the file as a parameter to launch Nemo, e.g. `nemo /path/to/file`. The challenge is that this uses a new window rather than existing one. That may or may not fit with the vision you have in mind.

EDIT: There is also an option `--existing-window`, e.g. `nemo --existing-window /path/to/file` that will create a new tab in an existing Nemo window if one is open or open a new window if not.

**Reply by anaximeno** [reply]
Hi Rick,

I was thinking of selecting only specific files or folders in the current window (maybe using regex), or activating the rename on a specific folder for example, for these cases that wouldn't be enough, I guess.

**Reply by rcalixte** [reply]
Maybe new switches for Nemo if there's enough general interest? I'm thinking something like `--regex-select` that would support selecting files and folders matching a regex parameter.

For renaming, maybe you could create a dialog window for the rename to workaround the limitation? Similar to how the bulk rename tools work.

**Reply by anaximeno** [reply]
Yeah, if a considerable amount of people show interest in that in the future it might be a good idea to add it by then.

The rename feature was for the move into a new folder action, I wanted to make it work the same way it works when you create a new folder, but anyway, I'll keep it the way it is working currently.

Thanks!
